src/main/java/org/apache/sling/commons/json/io/JSONWriter.java

```diff
@@ -243,134 +208,240 @@ public JSONWriter endObject() throws JSONException {
+            return value.toString();
+        }
+        if (value instanceof Map) {
+            Map<?, ?> map = (Map<?, ?>) value;
+            return new JSONObject(map).toString();
+        }
+        if (value instanceof Collection) {
+            Collection<?> coll = (Collection<?>) value;
+            return new JSONArray(coll).toString();
+        }
+        if (value.getClass().isArray()) {
+            return new JSONArray(value).toString();
+        }
+        if (value instanceof Enum<?>) {
+            return JSONObject.quote(((Enum<?>) value).name());
+        }
+        return JSONObject.quote(value.toString());
+    }
 
     /**
      * Append either the value <code>true</code> or the value
      * <code>false</code>.
+     * 
      * @param b A boolean.
      * @return this
-     * @throws JSONException
+     * @throws JSONException if a called function has an error
      */
     public JSONWriter value(boolean b) throws JSONException {
         return this.append(b ? "true" : "false");
     }
 
     /**
      * Append a double value.
+     * 
      * @param d A double.
      * @return this
      * @throws JSONException If the number is not finite.
      */
     public JSONWriter value(double d) throws JSONException {
-        return this.value(new Double(d));
+        return this.value(Double.valueOf(d));
     }
 
     /**
      * Append a long value.
+     * 
      * @param l A long.
      * @return this
-     * @throws JSONException
+     * @throws JSONException if a called function has an error
      */
     public JSONWriter value(long l) throws JSONException {
         return this.append(Long.toString(l));
     }
 
-
     /**
      * Append an object value.
-     * @param o The object to append. It can be null, or a Boolean, Number,
-     *   String, JSONObject, or JSONArray, or an object with a toJSONString()
-     *   method.
+     * 
+     * @param object The object to append. It can be null, or a Boolean, Number,
+     *               String, JSONObject, or JSONArray, or an object that implements
+     *               JSONString.
      * @return this
      * @throws JSONException If the value is out of sequence.
      */
-    public JSONWriter value(Object o) throws JSONException {
-        return this.append(JSONObject.valueToString(o));
+    public JSONWriter value(Object object) throws JSONException {
+        return this.append(valueToString(object));
+    }
+
+    /**
+     * @deprecated do not use, was removed from library and later stubbed
+     */
+    @Deprecated
+    public void setTidy(boolean tidy) {
+        // stubbed as this was removed from the library but still is referenced from
+        // customer code
+    }
+
+    /**
+     * @deprecated do not use, was removed from library and later stubbed
+     */
+    @Deprecated
+    public boolean isTidy() {
+        // stubbed as this was removed from the library but still is referenced from
+        // customer code
+        return true;
```

Comment:
Technically speaking this should return false, because the tidy mode is never enabled.

Reply:
my thought here, why true, was: in rare cases where a consumer of the json wants to display it on the UI for some reason (developer tools or whatever), they may check for tidyness and only show it when tidy. By assuming it to be tidy, it would at least be shown or further processed for the potential cases, while false could trigger a tidy function first, which isnt supported any longer so, depending on the client code, could end in a loop, as it never becomes tidy.

Reply:
see further below, done that.